homeassistant/components/solax/sensor.py

```diff
@@ -34,10 +49,29 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     async_track_time_interval(hass, endpoint.async_refresh, SCAN_INTERVAL)
     devices = []
     for sensor, (idx, unit) in api.inverter.sensor_map().items():
+        state_class = STATE_CLASS_MEASUREMENT
+        if sensor.startswith("Today"):
+            last_reset = None
```

Comment:
This will cause kWh/energy sensors of today to have a measurement state class, without a last_reset, making them current value sensors; that is not correct. As they represent a total value, not a current value.

Additionally, we have made some changes around this model, please adjust this PR so it lines up with these changes announced in this blog post:

<https://developers.home-assistant.io/blog/2021/08/16/state_class_total/>

Reply:
Done, the new STATE_CLASS_TOTAL_INCREASING makes it much cleaner to do what's needed here. Thanks!